usvm-core/src/main/kotlin/org/usvm/StepScope.kt

```diff
@@ -132,18 +145,32 @@ class StepScope<T : UState<Type, *, Statement, Context, *, T>, Type, Statement,
 
         // Interpret the first state as original and others as forked
         this.forkedStates += forkedStates.subList(1, forkedStates.size)
+
+        if (forkedStates.size > 1 || registerForkPoint) {
+            forkedStates.forEach { it.forkPoints += possibleForkPoint }
```

Comment:
Is it expected that `registerForkPoint = false` doesn't affect this line in case of the fork? If so, this should be highlighted in the documentation

Reply:
I renamed `registerForkPoint` to `alwaysRegisterForkPoint`. The logic behind fork points regitstration should be more clear now.